Ristretto cache is crashing when stopping the pipeline in streams mode (only since 4.14.0)

In order to start and stop the benthos pipeline on our integration tests we're running it in streams mode.
Since version 4.14.0, if we use a `ristretto` cache it will crash the pipeline and fail our integration tests. 
If we revert back to 4.13.0 or change the cache from `ristretto` to `memory` it will run successfully.

To reproduce the issue we can use the following config `pipeline.yaml`:

```yaml
input:
  generate:
    mapping: 'root = {"foo":"bar"}'

cache_resources:
- label: cache
  ristretto: {}
```

and the go file `main.go`:

```go
package main

import (
	"context"
	"os"
	"time"

	_ "github.com/benthosdev/benthos/v4/public/components/all"
	"github.com/benthosdev/benthos/v4/public/service"
)

func main() {
	content, _ := os.ReadFile("pipeline.yaml")
	builder := service.NewStreamBuilder()
	builder.SetYAML(string(content))
	stream, _ := builder.Build()
	go func() {
		stream.Run(context.Background())
	}()
	time.Sleep(time.Second)
	stream.Stop(context.Background())
}
```

if we use benthos `4.14.0` or `4.15.0` on `go.mod`:
```
module test

go 1.20

require (
	github.com/benthosdev/benthos/v4 v4.15.0
)
```

it will crash with `panic: send on closed channel`:
```
{"foo":"bar"}
panic: send on closed channel

goroutine 1 [running]:
github.com/dgraph-io/ristretto.(*Cache).Close(0xc0001ec000)
        /Users/amador/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/cache.go:359 +0x45
github.com/benthosdev/benthos/v4/internal/impl/dgraph.(*ristrettoCache).Close(0x1030f1940?, {0xc0011a8c68?, 0xc00157bbf0?})
        /Users/amador/go/pkg/mod/github.com/benthosdev/benthos/v4@v4.15.0/internal/impl/dgraph/cache_ristretto.go:155 +0x1d
github.com/benthosdev/benthos/v4/public/service.(*airGapCache).Close(0x100014736?, {0x104231528?, 0xc0000780c0?})
        /Users/amador/go/pkg/mod/github.com/benthosdev/benthos/v4@v4.15.0/public/service/cache.go:117 +0x2b
github.com/benthosdev/benthos/v4/internal/component/cache.(*metricsCache).Close(0x0?, {0x104231528?, 0xc0000780c0?})
        /Users/amador/go/pkg/mod/github.com/benthosdev/benthos/v4@v4.15.0/internal/component/cache/cache_metrics.go:135 +0x2b
github.com/benthosdev/benthos/v4/internal/manager.(*Type).WaitForClose(0xc000811340, {0x104231528, 0xc0000780c0})
        /Users/amador/go/pkg/mod/github.com/benthosdev/benthos/v4@v4.15.0/internal/manager/type.go:960 +0x378
github.com/benthosdev/benthos/v4/public/service.(*Stream).Stop(0xc0006e4000, {0x104231528, 0xc0000780c0})
        /Users/amador/go/pkg/mod/github.com/benthosdev/benthos/v4@v4.15.0/public/service/stream.go:166 +0x265
main.main()
        /Users/amador/My Drive/Development/Draft/tests/main.go:22 +0xde

Process finished with the exit code 2
```

If we change it back to `4.13.0` or change the cache from `ristretto` to `memory` it will run successfully:
```
{"foo":"bar"}

Process finished with the exit code 0

```

I'll try to debug it.

--- a/internal/impl/dgraph/cache_ristretto.go
+++ b/internal/impl/dgraph/cache_ristretto.go
@@ -69,6 +69,7 @@ type ristrettoCache struct {
 
 	retriesEnabled bool
 	boffPool       sync.Pool
+	closeOnce      sync.Once
 }
 
 func newRistrettoCache(defaultTTL time.Duration, retriesEnabled bool, backOff *backoff.ExponentialBackOff) (*ristrettoCache, error) {
@@ -152,6 +153,8 @@ func (r *ristrettoCache) Delete(ctx context.Context, key string) error {
 }
 
 func (r *ristrettoCache) Close(ctx context.Context) error {
-	r.cache.Close()
+	r.closeOnce.Do(func() {
+		r.cache.Close()
+	})
 	return nil
 }

